include/deal.II/matrix_free/util.h

```diff
@@ -0,0 +1,53 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+#ifndef dealii_matrix_free_util_h
+#define dealii_matrix_free_util_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/quadrature.h>
+
+#include <deal.II/simplex/quadrature_lib.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+namespace internal
+{
+  namespace MatrixFreeFunctions
+  {
+    template <int dim>
+    inline Quadrature<dim - 1>
+    get_face_quadrature(const Quadrature<dim> &quad)
+    {
+      if (dim == 2 || dim == 3)
+        for (unsigned int i = 1; i <= 3; ++i)
+          if (quad == Simplex::QGauss<dim>(i))
+            return Simplex::QGauss<dim - 1>(i);
+
+      Assert(false, ExcNotImplemented());
+
+      return Quadrature<dim - 1>();
+    }
```

Comment:
This is how I create face quadrature rules within `MatrixFree`. This is probably not a great solution!? What would be a better approach? Passing quadrature rules externally via `MatrixFree::reinit()`? My suggestion would be to postpone the decision after the time that mixed meshes are working and we now all actual requirements.

Reply:
I agree, we can postpone it for now.